src/Chunk.ts

```diff
@@ -593,6 +593,10 @@ export default class Chunk {
 						else magicString.addSource(new MagicString(rendered));
 					}
 				}
+				module.syntheticExports.forEach(s => {
+					const rendered = s.renderBlock(renderOptions);
+					magicString.addSource(new MagicString(rendered));
+				});
```

Comment:
Minor note: I would usually prefer a `for..of` loop over a `.forEach` because it offers about the same readability but is faster, especially on "unoptimized"  code, as I learned from a Google engineer. The reason is that it avoids jumping in and out of a function for each iteration.